src/mbgl/map/source.cpp

```diff
@@ -603,4 +606,58 @@ void Source::dumpDebugLogs() const {
     }
 }
 
+FeatureResults Source::featuresAt(const PrecisionPoint point, const uint8_t radius, const TransformState& transform) const {
+    // figure out tile (bounded by source max zoom)
+    LatLng p = transform.pointToLatLng(point);
+
+    const double sine = std::sin(p.latitude * M_PI / 180);
```

Comment:
At a glance, I think some of this math goes away entirely if we account for https://github.com/mapbox/mapbox-gl-native/pull/3172/files#discussion_r47417361.